tests/Validation/ValidationValidatorTest.php

```diff
@@ -5986,6 +5986,17 @@ public function testBeforeAndAfter()
 
         $v = new Validator($trans, ['x' => ['a' => ['v' => 'c']], 'y' => 'invalid'], ['x' => 'date', 'y' => 'date|before:x']);
         $this->assertTrue($v->fails());
+
+        $v = new Validator($trans, ['x' => '1970-01-01'], ['x' => 'nullable|date', 'y' => 'nullable|date|after:x']);
+        $this->assertTrue($v->passes());
```

Comment:
You're missing a test here (where `x` is explicitly `null`):

```php
$v = new Validator($trans, ['x' => null', 'y' => '1970-01-01'], ['x' => 'nullable|date', 'y' => 'nullable|date|after:x']);
$this->assertTrue($v->passes());
```

Reply:
this was also not passing before my  [PR #49871](https://github.com/laravel/framework/pull/49871)